db/document.go

```diff
@@ -76,12 +76,64 @@ type syncData struct {
 	removedRevisionBodyKeys map[string]string // keys of non-winning revisions that have been removed (and so may require deletion), indexed by revID
 }
 
+func (doc *SyncData) HashRedact(salt string) SyncData {
```

Comment:
Using 'doc' for *SyncData as it's usually used for 'document' in the code - would make it clearer to just use 'sd' or something.